pkg/upgrade/handle_upgrade.go

```diff
@@ -72,10 +73,13 @@ func (ctl *Controller) handlePlans(ctx context.Context) error {
 			if err != nil {
 				return objects, status, err
 			}
-			for _, node := range concurrentNodes {
+			concurrentNodeNames := make([]string, len(concurrentNodes))
+			for i := range concurrentNodes {
+				node := concurrentNodes[i]
 				objects = append(objects, upgradejob.New(obj, node, ctl.Name))
-				obj.Status.Applying = append(obj.Status.Applying, node.Name)
+				concurrentNodeNames[i] = upgradenode.Hostname(node)
 			}
+			obj.Status.Applying = concurrentNodeNames[:]
```

Comment:
The sort issue was spurious, this was where the dupes were coming from all along.